update.sh

```diff
@@ -33,23 +33,24 @@ hour=$(echo $hour | sed 's/^0*//')
 # Different desktop environment implementations
 case $XDG_CURRENT_DESKTOP in
 	Mint|Mate) setcmd="gsettings set org.mate.background picture-uri";;
-	Cinnamon) setcmd="gsettings set org.cinnamon.background picture-uri";;
-	*) setcmd="gsettings set org.gnome.desktop.background picture-uri";; # GNOME/Unity, default
+	MATE) setcmd="gsettings set org.mate.background picture-filename ";;
+	Cinnamon) setcmd="gsettings set org.cinnamon.background picture-uri file://";;
+	*) setcmd="gsettings set org.gnome.desktop.background picture-uri file://";; # GNOME/Unity, default
```

Comment:
Yes, at first I thought I was making some mistake, but my guess is that it might be version specific. Newer versions of the mate session manager set this to all-caps MATE (see this https://github.com/mate-desktop/mate-session-manager/blob/master/data/mate.desktop.in)

What distro/version are you using BTW? I might get a live-usb to test against it before I send another PR.

I'm thinking there might be a better way of detecting the session, but I'm not sure. I'll keep thinking about it.